zmon-controller-ui/js/controllers/AlertDetailsCtrl.js

```diff
@@ -23,6 +23,9 @@ angular.module('zmon2App').controller('AlertDetailsCtrl', ['$scope', '$location'
         $scope.showActiveAlerts = true; // 1st tab, 1st button
         $scope.showAlertsInDowntime = false; // 1st tab, 2nd button
         $scope.showCheckResults = false; // 1st tab, 3rd button
+        $scope.alertDetailsSearch = { // entity filter
+            str: localStorageService.get('alertDetailsSearchStr') || ''
```

Comment:
We need to $watch on it to update localStorage

Watching on a string didn't work and the recommended solution on many stackoverflow posts was to use an object property instead.

The reason being in this case that the input field exists inside a ui-bootstrap component (<tabset>) which defines its own isolated scope. The string will only pass its value, and not its reference. By using an object property it passes its reference, not its value.

http://stackoverflow.com/questions/27513462/angularjs-ngmodel-doesnt-work-inside-a-ui-bootstrap-tabset